docs/spark/spark-writes.md

```diff
@@ -326,28 +330,61 @@ USING iceberg
 PARTITIONED BY (days(ts), category)
 ```
 
-To write data to the sample table, your data needs to be sorted by `days(ts), category`.
+#### In Spark SQL
 
-If you're inserting data with SQL statement, you can use `ORDER BY` to achieve it, like below:
+To globally sort data based on `ts` and `category`:
 
 ```sql
 INSERT INTO prod.db.sample
 SELECT id, data, category, ts FROM another_table
 ORDER BY ts, category
 ```
 
-If you're inserting data with DataFrame, you can use either `orderBy`/`sort` to trigger global sort, or `sortWithinPartitions`
-to trigger local sort. Local sort for example:
+To locally sort data based on `ts` and `category`:
+
+```sql
+INSERT INTO prod.db.sample
+SELECT id, data, category, ts FROM another_table
+SORT BY ts, category
+```
+
+`SORT BY` clauses can also be used with partition transforms. The [date-and-timestamp-functions](https://spark.apache.org/docs/latest/sql-ref-functions-builtin.html#date-and-timestamp-functions) will be used when partition transforms is time related; truncate related functions such as `substr` will be used when partition transform is `truncate[W]`; [define and register UDFs](https://spark.apache.org/docs/latest/sql-ref-functions-udf-scalar.html) can also take effect.
+
+```sql
+INSERT INTO prod.db.sample
+SELECT id, data, category, ts FROM another_table
+SORT BY day(ts), category
+```
+
+#### In the Dataframe API
+
+To globally sort data based on `ts` and `category`:
+
+```scala
+data.sort("ts", "category")
+    .writeTo("prod.db.sample")
+    .append()
+```
+
+```scala
+data.orderBy("ts", "category")
+    .writeTo("prod.db.sample")
+    .append()
+```
+
+To locally sort data based on `ts` and `category`:
 
 ```scala
 data.sortWithinPartitions("ts", "category")
     .writeTo("prod.db.sample")
     .append()
 ```
 
-You can simply add the original column to the sort condition for the most partition transformations, except `bucket`.
+#### Bucket Transform
+
+It's easy to add the original column to the sort condition for the most partition transformations, except `bucket`.
```

Comment:
This is a personal thing, but I dislike the use of words like "easy" or "simple" in documentation. Computers are hard and I generally want folks to not feel like they don't understand something that is labeled as "easy" and to avoid this I try to never use "easy" or "simple". It also doesn't add much to this documentation.

For this sentence I would probably go with something like
`bucket` transforms require additional steps which are not necessary with other partition transforms.

If we want to keep this sentence then we should correct the grammar,
"It's easy to add the original column to the sort condition for ~the~ most partition transforms, except bucket"